src/redux/books/booksSlice.js

```diff
@@ -1,40 +1,60 @@
-import { createSlice } from '@reduxjs/toolkit';
-
-const initialState = [
-  {
-    item_id: 'item1',
-    title: 'The Great Gatsby',
-    author: 'John Smith',
-    category: 'Fiction',
-  },
-  {
-    item_id: 'item2',
-    title: 'Anna Karenina',
-    author: 'Leo Tolstoy',
-    category: 'Fiction',
-  },
-  {
-    item_id: 'item3',
-    title: 'The Selfish Gene',
-    author: 'Richard Dawkins',
-    category: 'Nonfiction',
-  },
-];
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+const GET_POST_BOOKS_URL = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/jFgFPX1ePQ3LgYOp0Krp/books';
+const fetchBooks = createAsyncThunk('books/fetchBooks', async () => {
+  const response = await axios.get(GET_POST_BOOKS_URL);
+  return response.data;
+});
+
+const addNewBook = createAsyncThunk('books/addNewBook', async (book) => {
+  await axios.post(GET_POST_BOOKS_URL, book);
+});
 
-export const booksSlice = createSlice({
-  name: 'books',
+const deleteBook = createAsyncThunk('books/deleteBook', async (id) => {
+  const DELETE_URL = `https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/jFgFPX1ePQ3LgYOp0Krp/books/${id}`;
+  await axios.delete(DELETE_URL, id);
+});
+
+const initialState = {
+  books: [],
+
+};
+
+const bookSlice = createSlice({
+  name: 'book',
   initialState,
   reducers: {
-    addBook: (state, action) => {
-      const { id, title, author } = action.payload;
-      state.push({ id, title, author });
-    },
-    removeBook: (state, action) => {
-      const idToRemove = action.payload;
-      return state.filter((book) => book.id !== idToRemove);
-    },
+    addBook: (state, action) => ({
+      ...state,
+      books: [...state.books, action.payload],
+    }),
+    removeBook: (state, action) => ({
+      ...state,
+      books: state.books.filter((book) => book.item_id !== action.payload),
+    }),
+  },
```

Comment:
- [ ] Good job connecting your bookstore to an API, 👏 , I noticed that the remove button is not working as it should , Kindly fix that to go to the next step 👍 

![Capture 1](https://user-images.githubusercontent.com/80458199/233791393-d6f80f43-b640-41b8-9e22-f9942337e19e.PNG)